### PR TITLE
fix: Display the current bids on Sale page (ignore Relay cache)

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Fix**

### Description
The auction team [has raised a problem](https://artsy.slack.com/archives/C02MU3DB3/p1648496726378999) that incorrect bids were displayed on the Sale page (the displayed value was different from the actual one. _Pay attention to the displayed bids value for Edgar Plans on demo video_). In order to display up-to-date information, the user had to manually update the data using pull-to-refresh. The problem was due to the fact that data from Relay Cache was displayed

### Useful links
* [Slack thread #1](https://artsy.slack.com/archives/CCS8TMQ0K/p1648497720511179)
* [Slack thread #2](https://artsy.slack.com/archives/C9SATFLUU/p1648657057885529)

#### Before
https://user-images.githubusercontent.com/3513494/161049427-bb39bd8a-7b37-44c2-b7fe-efac336528b0.mp4

#### After
https://user-images.githubusercontent.com/3513494/161049509-26b32ad6-5741-4907-a7ab-9bab517b69a1.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Display the current bids on Sale page - dimatretyak

</details>

### CODE REVIEW

**Review by olerichter00 — APPROVED** [review]
Thanks for the fix 🙇 

Do you think it would make sense to poll the data? [Relay 12 supports polling queries](https://relay.dev/docs/v12.0.0/api-reference/types/CacheConfig/). I'm just not sure if it's supported with `QueryRenderer` or if we would have to use hooks instead.

**Comment by dimatretyak** [comment]
> Do you think it would make sense to poll the data?

I think it makes sense. But this point should be discussed with others, perhaps such functionality is not needed 🤷‍♂️

**Review by dblandin — APPROVED** [review]
Let's give it a shot 👍🏼

File: src/app/Scenes/Sale/Sale.tsx [hunk 1]
```diff
@@ -412,8 +412,7 @@ export const SaleQueryRenderer: React.FC<{
               )
             }}
             cacheConfig={{
-              // Bypass Relay cache on retries.
-              ...(isRetry && { force: true }),
+              force: true,
             }}
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Sale/Sale.tsx:416]: @MounirDhahri I checked that too. But unfortunately staled data is returned anyway

### Some context
* a test account has been opened in incognito mode
* fetch policy strategy has been set in `store-and-network`

### Demo
https://user-images.githubusercontent.com/3513494/162018529-34f1251d-9673-4eda-8124-ec4dbcea942f.mp4

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/Sale/Sale.tsx:416]: thank you very much for the demo video  🙏

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/Sale/Sale.tsx:416]: this is good to find out now - we might need to check why it's broken

[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Sale/Sale.tsx:416]: > this is good to find out now - we might need to check why it's broken

I'll try to find out the reason tomorrow or the day after tomorrow. If there is progress, I will give feedback

[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Sale/Sale.tsx:416]: @MounirDhahri `cacheMiddleware` ignores `fetchPolicy` and for this reason the request will not be executed (even if `network-only` is specified). We can update [some checks](https://github.com/artsy/eigen/blob/main/src/app/relay/middlewares/cacheMiddleware.ts#L14) and ignore cache for specific fetch policies, but then another problem appears that we cannot get the specified `fetchPolicy` prop for `QueryRenderer` inside middleware 🤷‍♂️

In [this PR](https://github.com/artsy/eigen/pull/6572), I tried to solve the problem, but the solution turned out to be questionable (as it seems to me)

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/Sale/Sale.tsx:416]: thanks for following up on this. I will take a look at the other PR

**Pull request merged**